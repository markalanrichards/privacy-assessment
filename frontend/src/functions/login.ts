export interface LocalStorageFN<T> {
  setItem(index: string, item: T): void;
  getItem(index: string): T;
  clear(): void;
}
interface LoginInterface {
  username: string;
  status: string;
  login: boolean;
  error: null;
}
export const login = (localStorageFn: LocalStorageFN<string>) => (
  username: string
): LoginInterface => {
  localStorageFn.setItem("username", username);
  return {
    username: localStorageFn.getItem("username"),
    status: "success",
    login: true,
    error: null
  };
};

interface LogoutInterface {
  status: string;
  login: boolean;
  error: null;
}

export const logout = (
  localStorageFn: LocalStorageFN<string>
): LogoutInterface => {
  localStorageFn.clear();
  return {
    status: "logged out",
    login: false,
    error: null
  };
};

interface GetUserInterface {
  username: string;
  status: string;
  login: boolean;
  error: null;
}

export const getUser = (
  localStorageFn: LocalStorageFN<string>
): GetUserInterface => {
  const isTheUserLogedIn = localStorageFn.getItem("username");
  return isTheUserLogedIn === null || "undefined"
    ? {
        username: null,
        status: "No user logged in",
        login: false,
        error: null
      }
    : {
        username: isTheUserLogedIn,
        status: "User is logged in",
        login: true,
        error: null
      };
};
