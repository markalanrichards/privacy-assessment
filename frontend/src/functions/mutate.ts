interface MutateInterface<T> {
  externalLegalName: string;
  externalEmail: string;
}

export const mutate = (variables: MutateInterface<string>) => {
  return Promise.resolve({
    variables: {
      externalLegalName: variables.externalLegalName,
      externalEmail: variables.externalEmail
    }
  });
};
