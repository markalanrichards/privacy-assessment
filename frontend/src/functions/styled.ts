import * as styledComponents from "styled-components";

const {
  default: styled,
  css,
  injectGlobal,
  keyframes,
  ThemeProvider
} = styledComponents as styledComponents.ThemedStyledComponentsModule<
  IThemeInterface
>;

interface Color {
  primaryColor: string;
}

interface Size {
  large: string;
}

export interface IThemeInterface {
  color: Color;
  size: Size;
}

export const defaultTheme: IThemeInterface = {
  color: {
    primaryColor: "yellow"
  },
  size: {
    large: "10"
  }
};

export default styled;
export { css, injectGlobal, keyframes, ThemeProvider };
