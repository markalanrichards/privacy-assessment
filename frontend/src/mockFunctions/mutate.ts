import {
  SubjectProfileAvro,
  SubjectProfileCreateAvro,
  SubjectProfileUpdateAvro,
} from '../interfaces/SubjectProfile';

export const mutate = (
  variables:
    | SubjectProfileAvro
    | SubjectProfileCreateAvro
    | SubjectProfileUpdateAvro
) => {
  return Promise.resolve({
    variables,
  });
};
