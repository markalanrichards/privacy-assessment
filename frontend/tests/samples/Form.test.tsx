import * as React from "react";
import { CreateNewUserPage } from "./Form";
import { shallow } from "enzyme";
import * as mockApi from "../../src/functions/mutate";

const simulateChangeOnInput = (wrapper, inputSelector, newValue) => {
  const input = wrapper.find(inputSelector);
  input.simulate("change", {
    target: { value: newValue }
  });
  return wrapper.find(inputSelector);
};
describe("Form", () => {
  beforeEach(() => jest.resetAllMocks());
  it("lets me fill out the form with values", () => {
    const wrapper = shallow(<CreateNewUserPage mutate={mockApi.mutate} />);
    expect(wrapper.find("#externalLegalName").props().value).toEqual("");
    expect(wrapper.find("#externalEmail").props().value).toEqual("");
    const nameInput = simulateChangeOnInput(
      wrapper,
      "#externalLegalName",
      "James"
    );
    const emailInput = simulateChangeOnInput(
      wrapper,
      "#externalEmail",
      "james@test.com"
    );

    expect(nameInput.props().value).toEqual("James");
    expect(emailInput.props().value).toEqual("james@test.com");
    expect(wrapper).toMatchSnapshot();
  });

  it("lets me click the reset button to clear the form", () => {
    const wrapper = shallow(<CreateNewUserPage mutate={mockApi.mutate} />);
    const nameInput = simulateChangeOnInput(
      wrapper,
      "#externalLegalName",
      "James"
    );
    const emailInput = simulateChangeOnInput(
      wrapper,
      "#externalEmail",
      "james@test.com"
    );
    const resetBtn = wrapper.find('button[data-testid="reset-button"]');
    resetBtn.simulate("click");
    expect(wrapper.find("#externalLegalName").props().value).toEqual("");
    expect(wrapper.find("#externalEmail").props().value).toEqual("");
    expect(wrapper).toMatchSnapshot();
  });

  test("data is sumbited when button is pressed", () => {
    jest
      .spyOn(mockApi, "mutate")
      .mockImplementation(() => Promise.resolve({ message: "Saved user!" }));
    const wrapper = shallow(<CreateNewUserPage mutate={mockApi.mutate} />);

    const nameInput = simulateChangeOnInput(
      wrapper,
      "#externalLegalName",
      "James"
    );
    const emailInput = simulateChangeOnInput(
      wrapper,
      "#externalEmail",
      "james@test.com"
    );

    wrapper.find("form").simulate("submit", {
      preventDefault: () => {}
    });

    expect(mockApi.mutate).toHaveBeenCalledWith({
      variables: { externalEmail: "james@test.com", externalLegalName: "James" }
    });
  });
});
