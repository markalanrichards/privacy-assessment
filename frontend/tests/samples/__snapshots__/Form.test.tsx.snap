// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form lets me click the reset button to clear the form 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="group"
    >
      <label>
        Name
      </label>
      <input
        id="externalLegalName"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="group"
    >
      <label>
        Email
      </label>
      <input
        id="externalEmail"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <button
      data-testid="submit-button"
      type="submit"
    >
      Submit
    </button>
    <button
      data-testid="reset-button"
      onClick={[Function]}
      type="button"
    >
      Reset
    </button>
  </form>
</div>
`;

exports[`Form lets me fill out the form with values 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="group"
    >
      <label>
        Name
      </label>
      <input
        id="externalLegalName"
        onChange={[Function]}
        type="text"
        value="James"
      />
    </div>
    <div
      className="group"
    >
      <label>
        Email
      </label>
      <input
        id="externalEmail"
        onChange={[Function]}
        type="text"
        value="james@test.com"
      />
    </div>
    <button
      data-testid="submit-button"
      type="submit"
    >
      Submit
    </button>
    <button
      data-testid="reset-button"
      onClick={[Function]}
      type="button"
    >
      Reset
    </button>
  </form>
</div>
`;
